addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -372,14 +372,15 @@ export class ProductScreen extends Component {
         return Array.from(new Set([...barcodeMatches, ...fuzzyMatches]));
     }
 
-    getProductsByCategory(categoryId) {
-        const products = new Set(
-            this.pos.models["product.product"].getBy("pos_categ_ids", categoryId) || []
+    getProductsByCategory(category) {
+        let products = new Set(
```

Comment:
can you get the list of available categories and then filter the products that are part of these categories?
something along the lines of 
```js 
const allCategories = category.getAllChildren();
return this.pos.models["product.product"].filter(p => p.categ_ids.some(allCategories.includes))

Reply:
also, this is not necessarily part of this task, but can you make the `getAllChildren` method not take the category as a parameter? `mycateg.getAllChildren(mycateg)` is a little weird

Reply:
perhaps it should also a similar name to the one one the backend, maybe both should be called `getChildren` ( _get_children in python)